<launch>
    <arg name="model" />
    <param name="robot_description" textfile="$(find uav_simulator)/urdf/Avatar.urdf" />
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find uav_simulator)/config/show.rviz" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="son2" args="0 0 0 1.57 0 0 /base_link /laser_scan" output="screen" />
  <!-- simulator-->
  <node name="simulator" pkg="uav_simulator" type="simulator" output="screen">
    <!-- uav -->
    <param name="max_linear_velocity" value="1.0" />
    <param name="max_angular_velocity" value="1.0" />
    <param name="crash_limit" value="0.2" />
    <param name="arrive_limit" value="0.2" />
    <param name="angle_max" value="2.0943951023931953" />
    <param name="angle_min" value="-2.0943951023931953" />
    <param name="range_max" value="3.0" />
    <param name="range_min" value="0.15" />
    <param name="num_laser" value="40" />
    <param name="flight_height" value="0.5" />
    <param name="intergrate_dt" value="0.02" />
    <param name="accelerate_rate" value="1.0" />
    <param name="state_update_factor" value="0.1" />
    <!-- grid map -->
    <param name="target_distance" value="40" />
    <param name="safe_radius" value="0.25" />
    <param name="length_x" value="15" />
    <param name="length_y" value="50" />
    <param name="num_obs_max" value="20" />
    <param name="num_obs_min" value="2" />
    <param name="radius_obs_max" value="2.0" />
    <param name="radius_obs_min" value="0.25" />
    <param name="num_obs_max_extra" value="5" />
    <param name="num_obs_min_extra" value="2" />
    <param name="radius_obs_max_extra" value="0.3" />
    <param name="radius_obs_min_extra" value="0.25" />
    <param name="vibration_distance_extra" value="0.5" />
    <!-- tracking -->
    <param name="lead_distance_factor" value="3.0" />
    <param name="max_leading_distance" value="3.0" />
    <param name="min_leading_distance" value="1.0" />
    <param name="num_tracking_point" value="5" />
    <!-- reward -->
    <param name="distance_reward_allocation_factor" value="0.8" />
  </node>
</launch>
